### PR TITLE
feat: create comment and update functionality

### PR DESCRIPTION
**Pull request by nadduli**
**What does the PR do?**
This PR introduces new API endpoints to create and update comments. It includes server-side logic for handling comment creation and updates, along with appropriate validation and error handling.

**How should this be manually tested?**
1. **Create Comment**:
    - Send a POST request to `/api/v1/comments/add` with a valid `CreateCommentDto` payload and a valid user ID in the request.
    
    Example:
    ```json
    POST /api/v1/comments/add
    {
      "model_id": "1234",
      "model_type": "post",
      "content": "This is a comment"
    }
    ```

2. **Update Comment**:
    - Send a PATCH request to `/api/v1/comments/update/{id}` with a valid `UpdateCommentDto` payload and a valid user ID in the request.

    Example:
    ```json
    PATCH /api/v1/comments/update/1234
    {
      "content": "Updated comment content"
    }
    ```

**Checklist of what you did:**
- [x] Developed the service logic to create comments, including validation and error handling.
- [x] Developed the service logic to update comments, including validation and error handling.
- [x] Implemented the `addComment` endpoint to create comments.
- [x] Implemented the `updateComment` endpoint to update comments.
- [x] Added Swagger documentation for the new endpoints.
- [x] Handled various error scenarios, including missing content, user not found, and comment not found.
- [x] Wrote unit tests to verify that the endpoints correctly create and update comments.
- [x] Tested different scenarios to ensure the endpoints handle edge cases effectively.

**Reference Issue:**
[FEAT] Implement Create and Update Comment Functionality #368

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,7 +64,7 @@
   "devDependencies": {
     "@commitlint/cli": "^19.3.0",
     "@commitlint/config-conventional": "^19.2.2",
-    "@nestjs-modules/mailer": "^2.0.2",
+    "@nestjs-modules/mailer": "^1.6.1",
```
Comment (Cobidex) [inline package.json:67]: Why down grading the package?

Comment (nadduli) [reply package.json:67]: I. Will install packages using npm install

Comment (nadduli) [reply package.json:67]: I was installing packages using npm install but I get errors of deprecated packages
